python/helpers.py

```diff
@@ -0,0 +1,181 @@
+        self.key_coll = key_coll
+        self.key_vault_namespace = f"{self.key_db}.{self.key_coll}"
+        self.schema = schema
+        self.connection_string = connection_string
+        self.mongocryptd_bypass_spawn = mongocryptd_bypass_spawn
+        self.mongocryptd_spawn_path = mongocryptd_spawn_path
+
+    def ensure_unique_index_on_key_vault(self, key_vault):
+
+        # clients are required to create a unique partial index on keyAltNames
+        key_vault.create_index("keyAltNames",
+                               unique=True,
+                               partialFilterExpression={
+                                   "keyAltNames": {
+                                       "$exists": True
+                                   }
+                               })
+
+    def find_or_create_data_key(self):
+        """
+        In the guide, https://docs.mongodb.com/ecosystem/use-cases/client-side-field-level-encryption-guide/,
+        we create the data key and then show that it is created by
+        using a find_one query. Here, in implementation, we only create the key if
+        it doesn't already exist, ensuring we only have one local data key.
+
+        We also use the key_alt_names field and provide a key alt name to aid in
+        finding the key in the clients.py script.
+        """
+
+        key_vault_client = MongoClient(
+            self.connection_string, auto_encryption_opts=AutoEncryptionOpts(
+                self.kms_providers,
+                self.key_vault_namespace,
+                mongocryptd_bypass_spawn=self.mongocryptd_bypass_spawn,
+                mongocryptd_spawn_path=self.mongocryptd_spawn_path))
+
+        key_vault = (key_vault_client
+                     .get_database(self.key_db)
+                     .get_collection(self.key_coll))
+
+        self.ensure_unique_index_on_key_vault(key_vault)
+
+        data_key = key_vault.find_one(
+            {"keyAltNames": self.key_alt_name}
+        )
+
+        if data_key is None:
+            with ClientEncryption(self.kms_providers,
+                                  self.key_vault_namespace,
+                                  key_vault_client,
+                                  CodecOptions(uuid_representation=STANDARD)
+                                  ) as client_encryption:
+
+                data_key = client_encryption.create_data_key(
+                    "local", key_alt_names=[self.key_alt_name])
+                uuid_data_key_id = UUID(bytes=data_key)
+
+        else:
+            uuid_data_key_id = data_key["_id"]
+
+        base_64_data_key_id = (base64
+                               .b64encode(uuid_data_key_id.bytes)
+                               .decode("utf-8"))
+
+        return base_64_data_key_id
+
+    def get_regular_client(self):
+        return MongoClient(self.connection_string)
+
+    def get_csfle_enabled_client(self, schema=None):
+        if schema is None:
+            raise ValueError("schema is required")
```

Comment:
If schema is required can we just make it a required argument? `, schema=None)` -> `, schema)`